src/sinks/pulsar/service.rs

```diff
@@ -0,0 +1,201 @@
+use std::sync::Arc;
+use std::task::{Context, Poll};
+
+use bytes::Bytes;
+use futures::future::BoxFuture;
+use lru::LruCache;
+use pulsar::{Error as PulsarError, Executor, Producer, ProducerOptions, Pulsar};
+use tokio::sync::Mutex;
+use tower::Service;
+use vector_common::internal_event::CountByteSize;
+use vector_core::stream::DriverResponse;
+
+use crate::event::{EventFinalizers, EventStatus, Finalizable};
+use crate::internal_events::PulsarSendingError;
+use crate::sinks::pulsar::request_builder::PulsarMetadata;
+use crate::sinks::util::retries::RetryLogic;
+use std::num::NonZeroUsize;
+use vector_common::request_metadata::{MetaDescriptive, RequestMetadata};
+
+#[derive(Clone)]
+pub(super) struct PulsarRequest {
+    pub body: Bytes,
+    pub metadata: PulsarMetadata,
+    pub request_metadata: RequestMetadata,
+}
+
+pub struct PulsarResponse {
+    event_byte_size: usize,
+}
+
+impl DriverResponse for PulsarResponse {
+    fn event_status(&self) -> EventStatus {
+        EventStatus::Delivered
+    }
+
+    fn events_sent(&self) -> CountByteSize {
+        CountByteSize(1, self.event_byte_size)
+    }
+
+    fn bytes_sent(&self) -> Option<(usize, &str)> {
+        Some((self.event_byte_size, "pulsar"))
+    }
```

Comment:
This just recently changed to this:
```suggestion
    fn bytes_sent(&self) -> Option<usize> {
        Some(self.event_byte_size)
    }
```